### PR TITLE
[WORKFLOW] Actionlint + Fix mismatch

### PR DESCRIPTION
**Pull request by 0xdavinchee**

### CODE REVIEW

File: .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml [hunk 1]
```diff
@@ -31,16 +31,16 @@ jobs:
...
           if [ "$PUBLISHED_ETHEREUM_CONTRACTS_VERSION" != "$NEW_ETHEREUM_CONTRACTS_VERSION" ];then
-            echo "PUBLISH_ETHEREUM_CONTRACTS=1" >> $GITHUB_ENV
+            echo "PUBLISH_ETHEREUM_CONTRACTS=1" >> "$GITHUB_ENV"
             echo "ethereum-contracts changed: $PUBLISHED_ETHEREUM_CONTRACTS_VERSION -> $NEW_ETHEREUM_CONTRACTS_VERSION"
           else
             echo ethereum-contracts unchanged: $PUBLISHED_ETHEREUM_CONTRACTS_VERSION
           fi
 
-          echo "::set-output name=NEW_ETHEREUM_CONTRACTS_VERSION::$NEW_ETHEREUM_CONTRACTS_VERSION"
+          echo "name=NEW_ETHEREUM_CONTRACTS_VERSION" >> $NEW_ETHEREUM_CONTRACTS_VERSION
```
Comment (0xdavinchee) [inline .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml]: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/call.deploy-subgraph.yml [hunk 2]
```diff
@@ -35,12 +35,14 @@ jobs:
...
       - name: Show contexts
+        env:
+          HEAD_REF: ${{ github.head_ref }}
         run: |
           echo github.event_name: ${{ github.event_name }}
           echo github.sha: ${{ github.sha }}
           echo github.repository: ${{ github.repository }}
           echo github.ref: ${{ github.ref }}
-          echo github.head_ref: ${{ github.head_ref }}
+          echo github.head_ref: "$HEAD_REF"
```
Comment (hellwolf) [inline .github/workflows/call.deploy-subgraph.yml:46]: where is this suggestion coming from:

echo github.head_ref: "$HEAD_REF"

File: .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml [hunk 3]
```diff
@@ -31,16 +31,16 @@ jobs:
...
-            echo "PUBLISH_ETHEREUM_CONTRACTS=1" >> $GITHUB_ENV
+            echo "PUBLISH_ETHEREUM_CONTRACTS=1" >> "$GITHUB_ENV"
             echo "ethereum-contracts changed: $PUBLISHED_ETHEREUM_CONTRACTS_VERSION -> $NEW_ETHEREUM_CONTRACTS_VERSION"
           else
-            echo ethereum-contracts unchanged: $PUBLISHED_ETHEREUM_CONTRACTS_VERSION
+            echo ethereum-contracts unchanged: "$PUBLISHED_ETHEREUM_CONTRACTS_VERSION"
           fi
 
-          echo "::set-output name=NEW_ETHEREUM_CONTRACTS_VERSION::$NEW_ETHEREUM_CONTRACTS_VERSION"
+          echo "name=NEW_ETHEREUM_CONTRACTS_VERSION" >> "$NEW_ETHEREUM_CONTRACTS_VERSION"
```
Comment (hellwolf) [inline .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml]: typo?

**Review by hellwolf — CHANGES_REQUESTED** [review]

File: .github/workflows/call.deploy-subgraph.yml (hunk 2 above)
Comment (0xdavinchee) [reply .github/workflows/call.deploy-subgraph.yml:46]: from actionlint: https://files.slack.com/files-pri/TTSKFBES2-F04S9CF0P3R/screen_shot_2023-03-07_at_10.20.02_am.png
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections

Comment (0xdavinchee) [reply .github/workflows/call.deploy-subgraph.yml:46]: and a suggested fix: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable

File: .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml (hunk 3 above)
Comment (0xdavinchee) [reply .github/workflows/cd.ethereum-contracts-stable.create-release-drafts.yml]: ye, this one is a typo - is resolved now and double checked the other places for typos

File: .github/workflows/call.deploy-subgraph.yml (hunk 2 above)
Comment (hellwolf) [reply .github/workflows/call.deploy-subgraph.yml:46]: what about `github.ref`

Comment (0xdavinchee) [reply .github/workflows/call.deploy-subgraph.yml:46]: hmm, actionlint doesn't complain about it, but it is in that page, will add ref and any other things which might be an issue

File: .github/workflows/call.test-sdk-core.yml [hunk 4]
```diff
@@ -71,8 +71,11 @@ jobs:
 
       - name: "Run test suite"
         run: |
+          npx hardhat compile
+          # artificial slow down to give the subgraph time to sync
+          yarn build
```
Comment (hellwolf) [inline .github/workflows/call.test-sdk-core.yml]: do a explicitly sleep in the previous step.

**Review by hellwolf — APPROVED** [review]
one minor comments, otherwise looks good

**Pull request merged**